terraform/hcl/applicable/req_sw_terraform_aws_cloudfront_tls_only.tf

```diff
@@ -0,0 +1,22 @@
+resource "aws_cloudfront_distribution" "vulnerable_example" {
```

Comment:
[comment]: <> (FrogbotReview Iac 1 1 22 2 "vulnerable_example" /terraform/hcl/applicable/req_sw_terraform_aws_cloudfront_tls_only.tf)

## ![](https://raw.githubusercontent.com/jfrog/frogbot/master/resources/v2/applicableHighSeverity.png)<br>    high Infrastructure as Code Vulnerability 🛠️
	
Finding: viewer_protocol_policy='allow-all' was detected

### 👇 Details

Resource `aws_cloudfront_distribution` should have `viewer_protocol_policy=https-only` or `viewer_protocol_policy=redirect-to-https`

Vulnerable example -
```
resource "aws_cloudfront_distribution" "vulnerable_example" {
  default_cache_behavior {
    viewer_protocol_policy = "allow-all"
  }
}
```

Secure example -
```
resource "aws_cloudfront_distribution" "secure_example" {
  default_cache_behavior {
    viewer_protocol_policy = "https-only"
  }
}
```	

[[🐸 JFrog Frogbot]](https://github.com/jfrog/frogbot#readme)